runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/jobsubmission/InMemoryJobService.java

```diff
@@ -234,6 +237,26 @@ public void getState(
     }
   }
 
+  @Override
+  public void getPipeline(
+      GetJobPipelineRequest request, StreamObserver<GetJobPipelineResponse> responseObserver) {
+    LOG.trace("{} {}", GetJobPipelineRequest.class.getSimpleName(), request);
+    String invocationId = request.getJobId();
+    try {
+      JobInvocation invocation = getInvocation(invocationId);
+      RunnerApi.Pipeline pipeline = invocation.getPipeline();
```

Comment:
We aren't handling the NOT_FOUND case as described in the API contract in the proto.

Reply:
Again, I copied the API contract from `getState`, and handled the error case in exactly the same way, so if it's not correct for `getPipeline`, that's because it's not correct for `getState` or the other 4 methods that use this same pattern. 

In case of error it's responding with the exception:

```java
catch (Exception e) {
  ...
  responseObserver.onError(Status.INTERNAL.withCause(e).asException());
}
```

I'm new to Java, so I just assumed this was responding with `NOT_FOUND` in the case that `getInvocation(invocationId)` failed.

Reply:
We are handling the exception but are doing it incorrectly since getInvocation() throws a Status.NOT_FOUND exception but since we are catching it and then rethrowing it as an `INTERNAL` exception we are losing the `NOT_FOUND`.

The logic needs to be updated to be:
```
try {
  ... do stuff ...
} catch (StatusException e) {
  responseObserver.onError(e);
} catch (Exception e) {
  ... original exception handling that converts the status to INTERNAL ...
}
```

Reply:
Ok, I'll make a separate issue in Jira for this, since it's technically a breaking change, and I'll follow up with another PR with your suggestion, and the style fixes (if you don't mind me lumping them together).

Reply:
I would consider it a bugfix since contractually we said we were going to do X and did Y and not a breaking change.

I'm fine with you lumping it in or not.